src/ipyradiant/basic_tools/object_literal_collapsing.py

```diff
@@ -0,0 +1,156 @@
+# Copyright (c) 2020 ipyradiant contributors.
+# Distributed under the terms of the Modified BSD License.
+import traitlets as T
+
+import ipywidgets as W
+from rdflib import Graph, URIRef
+
+from .custom_uri_ref import CustomURIRef
+from .selection_widget import MultiPanelSelect
+
+
+class ObjectLiteralApp(W.VBox):
+    graph = T.Instance(Graph, allow_none=True)
+    lit_button = T.Instance(W.Button)
+    multiselect = T.Instance(MultiPanelSelect)
+    namespaces = T.Dict(default_value={}, allow_none=True)
+
+    def __init__(self, graph: Graph = None, *args, **kwargs):
+        """
+        This class is an example of how the MultiPanelSelect widget can be used in RDF
+        graph manipulation. This application combines that MultiPanelSelect widget with a graph
+        and a button that automatically moves certain predicates to the 'selected things' side via
+        a set of queries and set analysis.
+
+        This applciation inherits from the ipywidgets.VBox class as well.
+
+        params:
+        :graph: an rdflif.graph.Graph object, representing the graph to be queried on
+        :namespaces: a python dict object that represents the list of namespaces one wishes to use
+        with the current graph.
+        """
+        if graph is not None:
+            kwargs["graph"] = graph
+        if "layout" not in kwargs:
+            kwargs["layout"] = {
+                "min_height": "400px",
+                "flex": "1",
+            }
+        super().__init__(*args, **kwargs)
+
+        for key, value in self.namespaces.items():
+            self.graph.namespace_manager.bind(key, value)
+
+        self.lit_button.on_click(self.populate_predicates)
+        self.children = [self.multiselect, self.lit_button]
+
+    @T.default("graph")
+    def make_default_graph(self):
+        return Graph()
+
+    @T.default("lit_button")
+    def make_default_button(self):
+
+        layout = W.Layout(width="300px", height="40px")  # set width and height
+
+        button = W.Button(
+            description="Add predicates where object is a literal",
```

Comment:
The behavior of this button is getting close to complete. I noticed that adding a predicate (where the object is not a Literal) and then clicking the button removes the predicate that I added. We will want to fix that, but other than that it looks good. 

Let me know if you need additional clarification.